### DISCUSSION TITLE
Pagination for fetch all the users list of all organisation

### DISCUSSION THREAD

**Discussion by snehalsabade** [Sunbird-Lern/Community#132] [Query]
I'm trying to fetch all the users list of all organisation
This is the API https://sunbirdsaas.com/api/user/v1/search through which I'm fetching all users of all the organization.
{
    "request": {
        "filters": {
            "rootOrgId": []
        }
    }
}
In the response I'm getting about 3,60,000 users count.
I tried to add offset=9990 limit=10, but getting server side error
{
    "request": {
        "filters": {
            "rootOrgId": []
        },
        "limit":10,
        "offset":9991       
    }
}
I want to show all the 3.60,000 users in the UI through pagination but unable to achieve that.
Can anyone help me in this?

![Screenshot from 2023-06-14 15-33-33](https://github.com/Sunbird-Lern/Community/assets/132333617/1fd97c9f-cd22-4352-94e4-4aeb661ca6b4)

![Screenshot from 2023-06-14 15-32-35](https://github.com/Sunbird-Lern/Community/assets/132333617/b955edac-221b-4744-b39a-b6bd1159b074)

[@vinukumar-vs](https://github.com/vinukumar-vs) [@rajeevsathish](https://github.com/rajeevsathish) @[rajeshkumaravel](https://github.com/rajeshkumaravel) [@Krishnaj20](https://github.com/Krishnaj20)

**Comment by reshmi-nair** [comment]
@snehalsabade This is due to ES search limitation. Max records that can be fetched from ES by default is 10k records. Please refer : [ES doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html). 

Also its not good to fetch 3L + records in one API fetch and display it using pagination. It is better to use different filters like by name or by some specific details  and fetch the records to display. Otherwise the page rendering and performance will be very bad.

**Reply by Jayaprakash8887** [reply]
@snehalsabade Please mark this ticket as answered if there are no further queries.
